src/scenes/ObservationsRouter/details/index.tsx

```diff
@@ -132,7 +136,9 @@ export default function ObservationDetails(props: ObservationDetailsProps): JSX.
   const plantingSite = useAppSelector((state) => selectPlantingSite(state, plantingSiteId));
   const observation = useAppSelector((state) => selectObservation(state, plantingSiteId, observationId));
   const zoneNames = useAppSelector((state) =>
-    selectDetailsZoneNames(state, plantingSiteId, observationId, selectedOrganization)
+    !selectedOrganization?.id
+      ? []
+      : selectDetailsZoneNames(state, plantingSiteId, observationId, selectedOrganization.id)
```

Comment:
Passing the organization object instead of the ID seems like something the type system should have caught. (Fixing that is out of scope for this PR, of course.)

Reply:
I agree and was surprised it didn't catch it.

Reply:
I took a quick look and it seems like the answer is that `createCachedSelector` doesn't preserve argument type information.